connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/PluginClassLoader.java

```diff
@@ -106,6 +107,7 @@ public String toString() {
             if (resolve) {
                 resolveClass(klass);
             }
+            Plugins.compareAndSwapLoaders(oldClassloader);
```

Comment:
This should go in a `finally` block in case an exception is thrown, right?

Reply:
This has the same reasoning as the other lack of `finally`, and this is more performance critical since it's called on every classload in every plugin.

Reply:
This was actually a serious bug, so I pulled this up into the DelegatingClassLoader where we can actually afford a try-catch. I still couldn't implement the finally, but it doesn't have this bug any more.

Reply:
Mmm, makes sense. As long as the `DelegatingClassLoader` sets the current classloader to any `PluginClassLoader`s it delegates to before calling `loadClass` on them I don't think it should be necessary to also do comparing/swapping in the `PluginClassLoader` itself. Might want to double-check with the other reviewers, though.

Reply:
I've since removed the CAS from the `PluginClassLoader`, and it's now unaffected by this PR.